File: 0442 Find All Duplicates in an Array/Solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    var result = [];
    
    for (var i = 0; i < nums.length; ++ i) {
        var entry = Math.abs(nums[i]) - 1;
        if (nums[entry] < 0) {
            result.push(Math.abs(nums[i]));
        }
        nums[entry] *= -1;
    }
    
    return result;
};

File: 0442 Find All Duplicates in an Array/Solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        res = []

        for num in nums:
            absValue = abs(num)

            if nums[absValue-1] < 0:
                res.append(absValue)
            else:
                nums[absValue-1] *= -1

        return res
